### ISSUE TITLE
Incorrect type of starred argument in function definition

### ISSUE THREAD

**Issue by spkersten** [python/mypy#511]
The type of the starred argument is incorrectly inferred to be a `List` instead of a `Tuple`. For example:

``` python
def f(*y: int) -> None:
    p = 1, 2
    p = y
```

mypy incorrectly reports:

```
line 3: Incompatible types in assignment (expression has type List[int], variable has type "Tuple[int, int]")
```

See [documentation](https://docs.python.org/2/tutorial/controlflow.html#arbitrary-argument-lists).
